### ISSUE TITLE
Support UNNEST function in Continuous Aggregates

### ISSUE THREAD

**Issue by ruideyllot** [timescale/timescaledb#1717]
Hello!
I was trying to create a continuous aggregation view that would unnest the source information by a text array column but it seems to not be supported.

For example, suppose I have the following hypertable:
```
CREATE TABLE kpis_raw (
    time TIMESTAMP NOT NULL,
    value INTEGER,
    groups TEXT[]
);

INSERT INTO kpis_raw (time, value, groups)
VALUES
  (now(), 10, '{group1,group2,group3}'),
  (now(), 20, '{group1,group4}');
```
|time|value|groups|
| :---: | ---: | :--- |
| 2020-03-02 15:00:00 | 10 | {group1,group2,group3} |
| 2020-03-02 15:00:00 | 20 | {group1,group4} |

I can execute a normal query like this:
```
SELECT
  time_bucket('1m', kpis_raw."time") AS bucket,
  count(*) AS number_of_records,
  avg(value) AS average,
  unnest(groups) AS kpi_group
FROM kpis_raw
GROUP BY bucket, kpi_group;
```
|bucket|number_of_records|average|kpi_group|
| :---: | ---: | ---: | :---: |
| 2020-03-02 15:00:00 | 2 | 15 | group1 |
| 2020-03-02 15:00:00 | 1 | 10 | group2 |
| 2020-03-02 15:00:00 | 1 | 10 | group3 |
| 2020-03-02 15:00:00 | 1 | 20 | group4 |

But when I try to create this query as a continuous aggregation I get the following error:
```
ERROR:  invalid SELECT query for continuous aggregate
```
Is this really not supported as of version 1.6? Are there possible alternatives?

**Comment by nicolai-shape** [comment]
Is anything planned for this? It's not just `unnest`, it's any function that expands to multiple rows

**Comment by funny-falcon** [comment]
I don't get from documentation: LATERAL JOIN could be used for this?

**Comment by stalkerg** [comment]
@tucnak yes, currently, we should make a trigger to generate a new hypertable and, based on it create a mat view.

**Comment by tonyalaribe** [comment]
I ran into this today and have been wracking my head trying to workaround this. It would be great if we had some form of support for this.
